javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -528,15 +528,48 @@ object Character {
       throw new IllegalArgumentException()
 
     if (isSupplementaryCodePoint(codePoint)) {
-      val cpPrime = codePoint - 0x10000
-      val high = 0xD800 | ((cpPrime >> 10) & 0x3FF)
-      val low = 0xDC00 | (cpPrime & 0x3FF)
-      Array(high.toChar, low.toChar)
+      val dst = new Array[Char](2)
+      toSurrogate(codePoint, dst, 0)
+      dst
     } else {
       Array(codePoint.toChar)
     }
   }
 
+  def toChars(codePoint: Int, dst: Array[Char], dstIndex: Int): Int = {
+    if (!isValidCodePoint(codePoint))
+      throw new IllegalArgumentException()
+
+    if (dst == null)
+      throw new NullPointerException()
+
+    if (dstIndex < 0)
+      throw new ArrayIndexOutOfBoundsException(dstIndex.toString)
```

Comment:
These two tests are not necessary, nor is `checkIndex`. The accesses to `dst` inside `toSurrogate` will be erroneous and (in theory) cause the same exceptions anyway.

Of course NPEs and IndexOOBs are undefined behavior, so in practice the exceptions will not be thrown as is, but then also we don't care precisely because they are undefined behavior.

So just keep the `isValidCodePoint` test. Remove all the other checks.